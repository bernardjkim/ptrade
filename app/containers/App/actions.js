/*
 *
 * App actions
 *
 */

import {
  CREATE_SESSION,
  CREATE_SESSION_ERROR,
  CREATE_SESSION_SUCCESS,
} from './constants';

/**
 * Create session, this action starts the request saga
 *
 * @return {object} An action object with a type of CREATE_SESSION
 */
export function createSession() {
  return {
    type: CREATE_SESSION,
  };
}

/**
 * Dispatched when the session token is loaded by the request saga
 *
 * @param  {string} token The chart data
 *
 * @return {object}       An action object with a type of CREATE_SESSION_SUCCESS
 *                        passing the token
 */
export function createSessionSuccess(token) {
  return {
    type: CREATE_SESSION_SUCCESS,
    token,
  };
}

/**
 * Dispatched when createing the session fails
 *
 * @param  {object} error The error
 *
 * @return {object}       An action object with a type of CREATE_SESSION_ERROR
 *                        passing the error
 */
export function createSessionError(error) {
  return {
    type: CREATE_SESSION_ERROR,
    error,
  };
}
