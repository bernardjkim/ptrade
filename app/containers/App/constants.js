/*
 * AppConstants
 * Each action has a corresponding type, which the reducer knows and picks up on.
 * To avoid weird typos between the reducer and the actions, we save them as
 * constants here. We prefix them with 'yourproject/YourComponent' so we avoid
 * reducers accidentally picking up actions they shouldn't.
 *
 * Follow this format:
 * export const YOUR_ACTION_CONSTANT = 'yourproject/YourContainer/YOUR_ACTION_CONSTANT';
 */

// CREATE SESSION
export const CREATE_SESSION = 'app/App/CREATE_SESSION';
export const CREATE_SESSION_SUCCESS = 'app/App/CREATE_SESSION_SUCCESS';
export const CREATE_SESSION_ERROR = 'app/App/CREATE_SESSION_ERROR';

// DELETE SESSION
// TODO: is it enough to just delete the client-side token? or do i need to
// black list token on the backend?
export const DELETE_SESSION = 'app/App/DELETE_SESSION';
export const DELETE_SESSION_SUCCESS = 'app/App/DELETE_SESSION_SUCCESS';
export const DELETE_SESSION_ERROR = 'app/App/DELETE_SESSION_ERROR';

export const LOAD_TOKEN = 'app/App/LOAD_TOKEN';
