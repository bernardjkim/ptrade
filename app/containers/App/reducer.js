/*
 *
 * App reducer
 *
 */

import { fromJS } from 'immutable';
import {
  CREATE_SESSION,
  CREATE_SESSION_SUCCESS,
  CREATE_SESSION_ERROR,
} from './constants';

export const initialState = fromJS({
  token: false,
  loading: false,
  error: false,
});

function signinPageReducer(state = initialState, action) {
  switch (action.type) {
    case CREATE_SESSION:
      return state.set('loading', true).set('error', false);
    case CREATE_SESSION_SUCCESS:
      return state
        .set('token', action.token['Session-Token'])
        .set('loading', false);
    case CREATE_SESSION_ERROR:
      return state.set('error', action.error).set('loading', false);
    default:
      return state;
  }
}

export default signinPageReducer;
