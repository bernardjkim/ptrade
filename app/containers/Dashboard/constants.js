/*
 *
 * Dashboard constants
 *
 */

export const CHANGE_SEARCH = 'app/Dashboard/CHANGE_SEARCH';
export const CHANGE_TIME_FRAME = 'app/Dashboard/CHANGE_TIME_FRAME';

export const LOAD_STOCK_DATA = 'app/Dashboard/LOAD_STOCK_DATA';
export const LOAD_STOCK_DATA_SUCCESS = 'app/Dashboard/LOAD_STOCK_DATA_SUCCESS';
export const LOAD_STOCK_DATA_ERROR = 'app/Dashboard/LOAD_STOCK_DATA_ERROR';
