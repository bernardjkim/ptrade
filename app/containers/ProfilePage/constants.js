/*
 *
 * ProfilePage constants
 *
 */

export const CHANGE_TIME_FRAME = 'app/ProfilePage/CHANGE_TIME_FRAME';
export const SET_PORTFOLIO_VALUE = 'app/ProfilePage/SET_PORTFOLIO_VALUE';

export const LOAD_BALANCE = 'app/ProfilePage/LOAD_BALANCE';
export const LOAD_BALANCE_SUCCESS = 'app/ProfilePage/LOAD_BALANCE_SUCCCESS';
export const LOAD_BALANCE_ERROR = 'app/ProfilePage/LOAD_BALANCE_ERROR';

export const LOAD_POSITIONS = 'app/ProfilePage/LOAD_POSITIONS';
export const LOAD_POSITIONS_SUCCESS = 'app/ProfilePage/LOAD_POSITIONS_SUCCESS';
export const LOAD_POSITIONS_ERROR = 'app/ProfilePage/LOAD_POSITIONS_ERROR';

export const LOAD_CHART = 'app/ProfilePage/LOAD_CHART';
export const LOAD_CHART_SUCCESS = 'app/ProfilePage/LOAD_CHART_SUCCESS';
export const LOAD_CHART_ERROR = 'app/ProfilePage/LOAD_CHART_ERROR';
