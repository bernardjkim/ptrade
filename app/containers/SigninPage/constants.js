/*
 *
 * SigninPage constants
 *
 */

export const CHANGE_INPUT = 'app/SigninPage/CHANGE_INPUT';
