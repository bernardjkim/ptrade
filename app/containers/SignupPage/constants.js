/*
 *
 * SignupPage constants
 *
 */

export const CHANGE_INPUT = 'app/SignupPage/CHANGE_INPUT';

export const CREATE_USER = 'app/SignupPage/CREATE_USER';
export const CREATE_USER_SUCCESS = 'app/SignupPage/CREATE_USER_SUCCESS';
export const CREATE_USER_ERROR = 'app/SignupPage/CREATE_USER_ERROR';
