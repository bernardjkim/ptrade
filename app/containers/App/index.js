/**
 *
 * App.js
 *
 * This component is the skeleton around the actual pages, and should only
 * contain code that should be seen on all pages. (e.g. navigation bar)
 *
 */

import React from 'react';
import { Switch, Route } from 'react-router-dom';
import { compose } from 'redux';

import injectReducer from 'utils/injectReducer';
import injectSaga from 'utils/injectSaga';

import Dashboard from 'containers/Dashboard/Loadable';
// import HomePage from 'containers/HomePage/Loadable';
import SigninPage from 'containers/SigninPage/Loadable';
import SignupPage from 'containers/SignupPage/Loadable';
import NotFoundPage from 'containers/NotFoundPage/Loadable';
import ProfilePage from 'containers/ProfilePage/Loadable';
import saga from './saga';
import reducer from './reducer';

import GlobalStyle from '../../global-styles';

export function App() {
  return (
    <div>
      <Switch>
        <Route exact path="/" component={Dashboard} />
        <Route exact path="/dashboard" component={Dashboard} />
        <Route exact path="/signin" component={SigninPage} />
        <Route exact path="/signup" component={SignupPage} />
        <Route exact path="/profile" component={ProfilePage} />
        <Route component={NotFoundPage} />
      </Switch>
      <GlobalStyle />
    </div>
  );
}

const withReducer = injectReducer({ key: 'global', reducer });
const withSaga = injectSaga({ key: 'global', saga });

export default compose(
  withReducer,
  withSaga,
)(App);
