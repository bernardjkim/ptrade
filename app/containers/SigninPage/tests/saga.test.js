/**
 * Test sagas
 */

/* eslint-disable redux-saga/yield-effects */
import { takeLatest, put } from 'redux-saga/effects';
import { CREATE_SESSION } from 'containers/App/constants';
import {
  createSessionSuccess,
  createSessionError,
} from 'containers/App/actions';
import signinPageSaga, { getSession } from '../saga';

const email = 'test@email.com';
const password = 'password';

describe('createSession Saga', () => {
  let createSessionGenerator;

  // We have to test twice, once for a successful load and once for an unsuccessful one
  // so we do all the stuff that happens beforehand automatically in the beforeEach
  beforeEach(() => {
    createSessionGenerator = getSession();

    const selectDescriptor = createSessionGenerator.next().value;
    expect(selectDescriptor).toMatchSnapshot();

    createSessionGenerator.next(email);
    createSessionGenerator.next(password);
    const callDescriptor = createSessionGenerator.value;
    expect(callDescriptor).toMatchSnapshot();
  });

  it('request the token successfully', () => {
    const token = 'token';
    const putDescriptor = createSessionGenerator.next(token).value;
    expect(putDescriptor).toEqual(put(createSessionSuccess(token)));
  });

  it('should call the createSessionError action if the response errors', () => {
    const response = new Error('Some error');
    const putDescriptor = createSessionGenerator.throw(response).value;
    expect(putDescriptor).toEqual(put(createSessionError(response)));
  });
});

describe('Saga', () => {
  it('should start task to watch for SigninPage actions', () => {
    const takeLatestDescriptor = signinPageSaga().next().value;
    expect(takeLatestDescriptor).toEqual(
      takeLatest(CREATE_SESSION, getSession),
    );
  });
});
