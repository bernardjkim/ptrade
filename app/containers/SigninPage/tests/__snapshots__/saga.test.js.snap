// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`createSession Saga request the token successfully 1`] = `
Object {
  "@@redux-saga/IO": true,
  "SELECT": Object {
    "args": Array [],
    "selector": [Function],
  },
}
`;

exports[`createSession Saga request the token successfully 2`] = `undefined`;

exports[`createSession Saga should call the createSessionError action if the response errors 1`] = `
Object {
  "@@redux-saga/IO": true,
  "SELECT": Object {
    "args": Array [],
    "selector": [Function],
  },
}
`;

exports[`createSession Saga should call the createSessionError action if the response errors 2`] = `undefined`;
